libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -34,6 +35,8 @@ public class AdaptiveDialog : DialogContainer, IDialogDependencies
         public const string Kind = "Microsoft.AdaptiveDialog";
 
         internal const string ConditionTracker = "dialog._tracker.conditions";
+        internal const string GeneratorIdKey = "dialog.generatorId";
+        internal const string LanguagePolicyKey = "dialog.languagePolicy";
```

Comment:
Is using state the best way to communicate this?  Couldn't it be accessed through the this scope?  Tom would be the best to ask about this.

Reply:
We inject LG templates into expression engine, which is only exposed to `state`, not the whole `services`. 

So this change is to put everything LG evaluation needed into state.

Reply:
I'm just wondering if it should be in dialog rather than turn.  I believe turn is not persisted whereas dialog is.  Seems like this could be setup on every turn.

---
In reply to: [598778627](https://github.com/microsoft/botbuilder-dotnet/pull/5319#discussion_r598778627) [](ancestors = 598778627)

Reply:
Thanks, the reason it was put in dialog not turn, is this generatorKey is per dialog and this language policy is also per dialog(though inherited from a common base), so it's more like this dialog's property. 

To solve concern of persisting, i would prefer to switch to turn scope, rather than creating a new scope. It feels like a overkill to create a new scope, because though we are putting something LG needed to get evaluated, but it's not something LG special, it's just more dialog information needed to be exposed. That dialog information could be used somewhere else, that some actions may also need. What do you think?

Reply:
Can you please reach out to Tom to discuss?

Reply:
we should not be creating this LG Memory scope.